resolwe/flow/views/relation.py

```diff
@@ -93,11 +93,18 @@ def update(self, request, *args, **kwargs):
         the collection referenced in the ``Relation``.
         """
         instance = self.get_object()
-        if (
-            not request.user.has_perm("edit_collection", instance.collection)
-            and not request.user.is_superuser
-        ):
-            return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        if not request.user.is_superuser:
```

Comment:
Put this check into a separate method and reuse it in the function below, e.g. something like `self.has_change_permission(instance, request.user)`.